### PR TITLE
update repo readme with concepts from the paper

### PR DESCRIPTION
**Pull request by luisoala**
* added visual summary figure to docs/images/
* created a version of readme with
   * summary text and figure side by side
   * added the code snippet from the paper under "Trying it out" for an ML workflow example

### CODE REVIEW

**Comment by luisoala** [comment]
@benjelloun @marcenacp uploaded figure to docs/images and made a version with readme display. feel free to adjust

also added the ml workflow code snippet from the paper below the existing readme snippet that @mkuchnik already started

**Comment by luisoala** [comment]
* added link to and bibtex of the arxiv preprint

not quite sure what the best place to put it is, overall the readme could maybe use a spring cleaning (:

**Review by marcenacp — APPROVED** [review]
Great, thanks! Just have a few nits. Please, ignore the failing tests, they were fixed in a recent PR (Hugging Face changed their croissant endpoint)

File: README.md [hunk 1]
```diff
@@ -30,6 +30,21 @@ for x in ds.records(record_set="default"):
     print(x)
 ```
 
+Use it in your ML workflow:
+
+```python3
```
[the remark this answers is not in the document]
Comment (luisoala) [reply README.md:35]: yes i think a simple workflow like this is an accessible illustration for the average ML user landing here. @mkuchnik wdyt? any preferences?

[the remark this answers is not in the document]
Comment (mkuchnik) [reply README.md:35]: @luisoala Looks good to me!

**Review by benjelloun — APPROVED** [review]
Thanks, Luis!

**Pull request merged**